ospfd/ospf_ri.c

```diff
@@ -667,6 +680,289 @@ void ospf_router_info_update_sr(bool enable, struct sr_node *srn)
 	}
 }
 
+static void flush_ri_fad_subtlvs(struct ri_fad_tlv *fad_tlv)
+{
+	struct tlv *subtlv;
+
+	FOREACH_TLV_IN_LIST (&fad_tlv->sub_tlvs, subtlv) {
+		tlv_list_del(&fad_tlv->sub_tlvs, subtlv);
+
+		switch (ntohs(subtlv->hdr.type)) {
+		case RI_FAD_EXC_ADMINGRP_SUBTLV:
+			/* SubTLV-specific cleanup */
+			XFREE(MTYPE_OSPF_RI_FAD_EXC_ADMNGRP_SUBTLV, subtlv);
+			break;
+		case RI_FAD_INCANY_ADMINGRP_SUBTLV:
+			/* SubTLV-specific cleanup */
+			XFREE(MTYPE_OSPF_RI_FAD_INCANY_ADMNGRP_SUBTLV, subtlv);
+			break;
+		case RI_FAD_INCALL_ADMINGRP_SUBTLV:
+			/* SubTLV-specific cleanup */
+			XFREE(MTYPE_OSPF_RI_FAD_INCALL_ADMNGRP_SUBTLV, subtlv);
+			break;
+		case RI_FAD_FLAGS_SUBTLV:
+			/* SubTLV-specific cleanup */
+			XFREE(MTYPE_OSPF_RI_FAD_FLAGS_SUBTLV, subtlv);
+			break;
+		case RI_FAD_EXC_SRLG_SUBTLV:
+			/* SubTLV-specific cleanup */
+			XFREE(MTYPE_OSPF_RI_FAD_EXC_SRLG_SUBTLV, subtlv);
+			break;
+		default:
+			assert(0);
```

Comment:
do we really need to crash here? could we print an error and ignore the TLV? if we do need to crash here, can we add more information?

Reply:
I think this is for a locally allocated memory in flex-also DB. We should never hit the assert(0). But I will remove it anyhow. The assert is more of for catching memory allocations of wrong types or more likely memory corruption.